OpenRA.Mods.Common/Activities/Air/HeliReturnToBase.cs

```diff
@@ -1,135 +0,0 @@
-#region Copyright & License Information
-/*
- * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
- * This file is part of OpenRA, which is free software. It is made
- * available to you under the terms of the GNU General Public License
- * as published by the Free Software Foundation, either version 3 of
- * the License, or (at your option) any later version. For more
- * information, see COPYING.
- */
-#endregion
-
-using System.Collections.Generic;
-using System.Linq;
-using OpenRA.Activities;
-using OpenRA.Mods.Common.Traits;
-using OpenRA.Traits;
-
-namespace OpenRA.Mods.Common.Activities
-{
-	public class HeliReturnToBase : Activity
-	{
-		readonly Aircraft aircraft;
-		readonly RepairableInfo repairableInfo;
-		readonly Rearmable rearmable;
-		readonly bool alwaysLand;
-		readonly bool abortOnResupply;
-		Actor dest;
-
-		public HeliReturnToBase(Actor self, bool abortOnResupply, Actor dest = null, bool alwaysLand = true)
-		{
-			aircraft = self.Trait<Aircraft>();
-			repairableInfo = self.Info.TraitInfoOrDefault<RepairableInfo>();
-			rearmable = self.TraitOrDefault<Rearmable>();
-			this.alwaysLand = alwaysLand;
-			this.abortOnResupply = abortOnResupply;
-			this.dest = dest;
-		}
-
-		public Actor ChooseResupplier(Actor self, bool unreservedOnly)
-		{
-			if (rearmable == null)
```

Comment:
`ReturnToBase`'s `ChooseResupplier` does a TraitInfo query instead of checking against `rearmable` - GH won't let me comment there directly.